packages/manager/apps/key-management-service/src/modules/secret-manager/pages/drawers/createVersionDrawer/CreateVersionDrawer.page.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { Suspense } from 'react';
+import z from 'zod';
+import { Drawer, ErrorBanner } from '@ovh-ux/manager-react-components';
+import { useTranslation } from 'react-i18next';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useForm } from 'react-hook-form';
+import { OdsMessage } from '@ovhcloud/ods-components/react';
+import { useSecretDataSchema } from '@secret-manager/validation';
+import { NAMESPACES } from '@ovh-ux/manager-common-translations';
+import { zodResolver } from '@hookform/resolvers/zod/dist/zod';
+import { SecretWithData } from '@secret-manager/types/secret.type';
+import { useSecretWithData } from '@secret-manager/data/hooks/useSecret';
+import { decodeSecretPath } from '@secret-manager/utils/secretPath';
+import { useCreateSecretVersion } from '@secret-manager/data/hooks/useCreateSecretVersion';
+import { SECRET_MANAGER_ROUTES_URLS } from '@secret-manager/routes/routes.constants';
+import { SecretDataFormField } from '@secret-manager/components/form/SecretDataFormField.component';
+import {
+  DrawerContent,
+  DrawerFooter,
+} from '@/common/components/drawer/DrawerInnerComponents.component';
+
+type CreateVersionDrawerProps = {
+  secret: SecretWithData;
+  domainId: string;
+  secretPath: string;
+  onDismiss: () => void;
+};
+
+const CreateVersionDrawerForm = ({
+  secret,
+  domainId,
+  secretPath,
+  onDismiss,
+}: CreateVersionDrawerProps) => {
+  const { t } = useTranslation([
+    'secret-manager/dashboard',
+    NAMESPACES.ACTIONS,
+  ]);
+
+  const {
+    mutateAsync: createSecretVersion,
+    isPending: isCreating,
+    error: createError,
+  } = useCreateSecretVersion();
+
+  const dataSchema = useSecretDataSchema();
+  const formSchema = z.object({ data: dataSchema });
+  type FormSchema = z.infer<typeof formSchema>;
+  const {
+    control,
+    handleSubmit,
+    formState: { isDirty, isValid },
+  } = useForm({
+    mode: 'onChange', // Validation on all changes to show errors immediately
```

Comment:
```suggestion
    mode: 'onTouched',
```
on some cases (current version is deactivated) it showed an error on init, which is not a great UX.